core/src/components/datetime/datetime.tsx

```diff
@@ -2393,6 +2388,12 @@ export class Datetime implements ComponentInterface {
                       isAdjacentDay,
                     });
 
+                    if (isAdjacentDay) {
+                      // The user selected a day outside the current month. Ignore this button, as the month will be re-rendered.
+                      this.el.blur();
+                      this.animateToDate(this.workingParts);
+                    }
```

Comment:
This isn't as smooth as when you press the arrow keys. There's a minor glitch after the animation ends. In the following video, it shows a smooth transition without readjustment when using the arrows. But when I select an adjacent day, the month does update smoothly but the dates jump around. That shouldn't be the case, it should be the same as when a user clicks on the arrows.


https://github.com/user-attachments/assets/82acd72e-475e-4398-957f-503dbde07187